src/usbg_common.c

```diff
@@ -359,6 +359,60 @@ int usbg_get_dev(const char *path, const char *name, const char *attr,
 	return 0;
 }
 
+int usbg_get_config_node_os_desc(config_setting_t *root, const char *iname,
+				struct usbg_function_os_desc *f_os_desc)
+{
+	const char *str_addr;
+	int ret;
+
+	ret = usbg_get_config_node_string(root, "compatible_id", &str_addr);
+	/* if error */
+	if (ret < 0)
+		return ret;
+
+	if (ret) {
+		strncpy(f_os_desc->compatible_id, str_addr, USBG_MAX_STR_LENGTH);
+		f_os_desc->compatible_id[USBG_MAX_STR_LENGTH - 1] = '\0';
+	}
+
+	ret = usbg_get_config_node_string(root, "sub_compatible_id", &str_addr);
+	/* if error */
+	if (ret < 0)
+		return ret;
+
+	if (ret) {
+		strncpy(f_os_desc->sub_compatible_id, str_addr, USBG_MAX_STR_LENGTH);
+		f_os_desc->sub_compatible_id[USBG_MAX_STR_LENGTH - 1] = '\0';
+	}
+
+	return 1;
```

Comment:
1?

Reply:
Hm, I think the idea was to indicate whether we have strings.. But seems buggy too.

Anyway, now that I use pointers in usbg_function_os_desc, we can use NULL to indicate a missing value and just not set it in usbg_set_interf_os_desc.